javalib/src/main/scala/java/util/concurrent/Semaphore.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.concurrent
+
+import java.util.{Collection, Collections}
+
+class Semaphore(private[this] var permits: Int, fairness: Boolean) extends Serializable {
+
+  def this(permits: Int) = this(permits, false)
+
+  // These methods can’t be implemented because they block
+  // def acquire(): Unit
+  // def acquire(permits: Int): Unit
+  // def acquireUninterruptibly(): Unit
+  // def acquireUninterruptibly(permits: Int): Unit
+  // def tryAcquire(permits: Int, timeout: Long, unit: TimeUnit): Boolean
+  // def tryAcquire(timeout: Long, unit: TimeUnit): Boolean
+
+  def availablePermits(): Int = permits
+
+  def drainPermits(): Int = {
+    val old = permits
+    permits = 0
+    old
+  }
+
+  protected def getQueuedThreads(): Collection[Thread] = Collections.emptySet()
+
+  @inline private def requireNonNegative(n: Int): Unit = {
+    if (n < 0) {
+      throw new IllegalArgumentException()
```

Comment:
I don't think we put braces on zero arg constructors. @sjrd?

Reply:
We don't have a definite rule for this. Either is fine. I don't think I am even consistent with myself about this.